var/spack/repos/builtin/packages/oras/package.py

```diff
@@ -0,0 +1,47 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import platform
+
+import llnl.util.tty as tty
+
+from spack import *
+
+
+class Oras(Package):
+    """ORAS means OCI Registry As Storage"""
+
+    homepage = "https://oras.land"
+    git      = "https://github.com/oras-project/oras"
+    url      = "https://github.com/oras-project/oras/archive/refs/tags/v0.12.0.tar.gz"
+
+    maintainers = ['vsoch']
+
+    version('main', branch="main")
+    version("0.12.0", sha256="5e19d61683a57b414efd75bd1b0290c941b8faace5fcc9d488f5e4aa674bf03e")
+
+    depends_on("go", type='build')
+
+    def setup_build_environment(self, env):
+        # Point GOPATH at the top of the staging dir for the build step.
+        env.prepend_path('GOPATH', self.stage.path)
+
+    def install(self, spec, prefix):
+        if platform.system() == "Linux" and platform.machine() == 'aarch64':
+            make("build-linux-arm64")
+        elif platform.system() == "Linux":
+            make("build-linux")
+        elif platform.system() == "Darwin" and platform.machine() == 'aarch64':
+            make("build-mac-arm64")
+        elif platform.system() == "Darwin":
+            make("build-mac")
+        elif platform.system() == "Windows":
+            make("build-windows")
```

Comment:
Instead of using Python `platform` can we impose the condition on the spec platform / target ?

Reply:
Python `platform` tests the current host. What you can do is to leverage the spec syntax:
```python
# if platform.system() == "Linux" and platform.machine() == 'aarch64':
if self.spec.satisfies('platform=linux target=aarch64:'):
```
so you don't have to chase all the nuances of different Python interpreters (iirc one of which is `arm64` vs. `aarch64`).

Reply:
ah gotcha! I'll try that.

Reply:
okay all set!